test/unit/mp_error.cc

```diff
@@ -349,7 +349,7 @@ test_decode_unknown_type()
 	const char *pos = buffer;
 	struct error *unpacked = error_unpack(&pos, len);
 	error_ref(unpacked);
-	err.code = 0;
+	err.code = 1;
```

Comment:
Thanks for paying attention! However nope, it is related. The encoded error has code 1 and the code was expected to be lost because the error type `"SomeNewError"` is unknown. But after my changes the code does not depend on error type and is always decoded and set. Hence in `error_unpack` you now get a correct code.